### PR TITLE
Tracing: Allows setting headers for OTel exporter

### PR DESCRIPTION
**Pull request by anuraaga**
# Description

Adds a configuration setting for OTel to set header key/value pairs to include in export requests. This can be used to add e.g. API keys for vendor tracing backends.

## Issue reference

Please reference the issue this PR will close: #8143

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [X] Created/updated tests
- [X] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Specification has been updated / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_
- [ ] Provided sample for the feature / Created issue in the <https://github.com/dapr/docs/> repo: dapr/docs#_[issue number]_

I will add either issue or PR in docs repo after an initial review confirming the general approach / feature is fine.

I couldn't find any e2e-type of tests for checking the OTel exporter configuration so added just some simple unit tests. Let me know if there is anything I should update, otherwise I tried referring to PRs to the code area to see the patterns for testing.

/cc @codefromthecrypt

### CODE REVIEW

File: pkg/config/configuration.go [hunk 1]
```diff
@@ -242,7 +242,8 @@ type OtelSpec struct {
 	Protocol        string `json:"protocol,omitempty" yaml:"protocol,omitempty"`
 	EndpointAddress string `json:"endpointAddress,omitempty" yaml:"endpointAddress,omitempty"`
 	// Defaults to true
-	IsSecure *bool `json:"isSecure,omitempty" yaml:"isSecure,omitempty"`
+	IsSecure *bool             `json:"isSecure,omitempty" yaml:"isSecure,omitempty"`
+	Headers  map[string]string `json:"headers,omitempty" yaml:"headers,omitempty"`
```
Comment (anuraaga) [inline pkg/config/configuration.go:246]: The name `headers` follows the same from OTel collector

https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configgrpc/README.md#client-configuration

File: pkg/config/env/env_variables.go [hunk 2]
```diff
@@ -23,4 +23,6 @@ const (
 	OtlpExporterInsecure string = "OTEL_EXPORTER_OTLP_INSECURE"
 	// OpenTelemetry transport protocol (grpc, http/protobuf, http/json)
 	OtlpExporterProtocol string = "OTEL_EXPORTER_OTLP_PROTOCOL"
+	// OpenTelemetry headers to add to transport
+	OTEL_EXPORTER_OTLP_HEADERS string = "OTEL_EXPORTER_OTLP_HEADERS"
```
Comment (anuraaga) [inline pkg/config/env/env_variables.go:27]: env var defined here

https://opentelemetry.io/docs/specs/otel/protocol/exporter/#specifying-headers-via-environment-variables

File: pkg/config/configuration_test.go [hunk 3]
```diff
@@ -479,6 +479,7 @@ func TestSetTracingSpecFromEnv(t *testing.T) {
 	t.Setenv("OTEL_EXPORTER_OTLP_ENDPOINT", "http://otlpendpoint:1234")
 	t.Setenv("OTEL_EXPORTER_OTLP_INSECURE", "true")
 	t.Setenv("OTEL_EXPORTER_OTLP_PROTOCOL", "http/json")
+	t.Setenv("OTEL_EXPORTER_OTLP_HEADERS", "authorization=Bearer token,  User-Agent = dapr,x-api-key= 1234 ,x-debug =no")
```
[the remark this answers is not in the document]
Comment (cicoyle) [reply pkg/config/configuration_test.go:482]: Or was this was intentional based on the space trimming from the configuration.go file changes?

**Comment by cicoyle** [comment]
@anuraaga - mind running `make lint` locally and fixing the lint issues?

File: pkg/config/env/env_variables.go (hunk 2 above)
Comment (antontroshin) [reply pkg/config/env/env_variables.go:27]: The lint warning relates to the variable name `OTEL_EXPORTER_OTLP_HEADERS string`, should be `OtlpExporterHeaders string`

**Review by anuraaga — COMMENTED** [review]
Sorry about that! When I run `make lint` I get many linting errors across unrelated files so couldn't find the issues with these files, I guess CI is only checking the diff so has an easier time.

**Comment by anuraaga** [comment]
Thanks all - I saw @antontroshin's PR which seems to have some additional features and brings some more validation from the collector I missed. I'll go ahead and close this one since that should take care of things.

**Pull request closed without merging**